library/spdm_responder_lib/measurements.c

```diff
@@ -301,12 +299,29 @@ return_status spdm_get_response_measurements(IN void *context,
 		measurements,
 		&measurements_size);
 
-	if (!ret) {
-		spdm_generate_error_response(spdm_context,
-					     SPDM_ERROR_CODE_UNSPECIFIED,
-					     0, response_size, response);
-		return RETURN_SUCCESS;
+

```

Comment:
I'm a bit unclear on when we should be returning errors and returning success combined with calling spdm_generate_error_response. It seem that sometimes we return errors and set an error response and other times we return success and set an error response. We should clear that up.

Reply:
We should make it consistent in this patch set.

If we want to change, I prefer we change it together with all other response function.

Reply:
Ok. Then on error, I will go with existing convention and make the spdm_get_response_measurements return RETURN_SUCCESS and error will be generated via spdm_generate_error_response.  There was one piece of code that I did not modify that return RETURN_UNSUPPORTED and calls spdm_generate_error_response, which is what confused me (see below). All other errors return RETURN_SUCCESS when calling spdm_generate_error_reponse. Should this one below be changed to RETURN_SUCCESS?

		session_state = spdm_secured_message_get_session_state(
			session_info->secured_message_context);
		if (session_state != SPDM_SESSION_STATE_ESTABLISHED) {
			spdm_generate_error_response(
				spdm_context,
				SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
				response_size, response);
			return RETURN_UNSUPPORTED;
		}